newrelic/hooks/framework_azurefunctions.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import urllib.parse as urlparse
+
+from newrelic.api.application import application_instance
+from newrelic.api.transaction import current_transaction
+from newrelic.api.web_transaction import WebTransaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
+from newrelic.common.signature import bind_args
+from newrelic.common.utilization import AZURE_RESOURCE_GROUP_NAME_PARTIAL_RE, AZURE_RESOURCE_GROUP_NAME_RE
+
+
+def original_agent_instance():
+    app_name = os.environ.setdefault("NEW_RELIC_APP_NAME", os.getenv("WEBSITE_SITE_NAME", ""))
+
+    return application_instance(app_name)
+
+
+def intrinsics_populator(application, context):
+    # For now, only HTTP triggers are supported
+    trigger_type = "http"
+
+    website_owner_name = os.environ.get("WEBSITE_OWNER_NAME", None)
```

Comment:
Perhaps we could throw this whole block into a try-except and log a warning given it looks like there are a few potential failure points here. Or we could ideally minimize the direct attribute access like `context.function_name` and `application.name` and use `getattr` calls instead

Reply:
Good point--Done!